### PR TITLE
re-enforce type guard

### PR DESCRIPTION
**Pull request by PorterNan**
# What

Sometimes AreEqual will mix up those buttons with all optional props(Especially for cameraButton), this change will also compare all optional props and parameters to make sure AreEqual is working properly

# Why

# How Tested

# Process & policy checklist

- [ ] I have updated the project documentation to reflect my changes if necessary.
- [ ] I have read the [CONTRIBUTING](https://github.com/Azure/communication-ui-library/blob/main/CONTRIBUTING.md) documentation.

**Is this a breaking change?**

- [ ] This change causes current functionality to break.

### CODE REVIEW

File: packages/acs-ui-common/review/acs-ui-common.api.md [hunk 1]
```diff
@@ -7,7 +7,13 @@
 import { CommunicationIdentifier } from '@azure/communication-common';
 
 // @public (undocumented)
-export type AreEqual<A, B> = A extends B ? (B extends A ? true : false) : false;
+export type AreEqual<A extends (props: any) => JSX.Element | undefined, B extends (props: any) => JSX.Element | undefined> = true extends AreTypeEqual<A, B> & AreParamEqual<A, B> ? true : false;
+
+// @public (undocumented)
+export type AreParamEqual<A extends (props: any) => JSX.Element | undefined, B extends (props: any) => JSX.Element | undefined> = AreTypeEqual<Required<Parameters<A>[0]>, Required<Parameters<B>[0]>>;
```
Comment (JamesBurnside) [inline packages/acs-ui-common/review/acs-ui-common.api.md:13]: What's the difference between this and AreEqual?

**Review by JamesBurnside — APPROVED** [review]

File: packages/acs-ui-common/review/acs-ui-common.api.md [hunk 2]
```diff
@@ -7,7 +7,13 @@
 import { CommunicationIdentifier } from '@azure/communication-common';
 
 // @public (undocumented)
-export type AreEqual<A, B> = A extends B ? (B extends A ? true : false) : false;
+export type AreEqual<A extends (props: any) => JSX.Element | undefined, B extends (props: any) => JSX.Element | undefined> = true extends AreTypeEqual<A, B> & AreParamEqual<A, B> ? true : false;
```
Comment (JamesBurnside) [inline packages/acs-ui-common/review/acs-ui-common.api.md:10]: Could the `extends (props: any) => JSX.Element | undefined` be something like `extends React.Component`?

Comment (PorterNan) [reply packages/acs-ui-common/review/acs-ui-common.api.md:10]: Good catch! React.Component contains more than just functional component, which makes this expression a little bit tricky

**Review by alkwa-msft — APPROVED** [review]

**Pull request merged**